libs/cache/file_cache.go

```diff
@@ -0,0 +1,250 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/databricks/cli/internal/build"
+	"github.com/databricks/cli/libs/log"
+)
+
+// Metrics is a local interface for tracking cache telemetry.
+type Metrics interface {
+	SetBoolValue(key string, value bool)
+}
+
+// FileCache implements the Cache interface using local disk storage.
+type FileCache[T any] struct {
+	baseDir       string
+	expiryMinutes int
+	mu            sync.Mutex
+	metrics       Metrics
+}
+
+// newFileCacheWithBaseDir creates a new file-based cache that stores data in the specified directory.
+func newFileCacheWithBaseDir[T any](baseDir string, expiryMinutes int) (*FileCache[T], error) {
+	if err := os.MkdirAll(baseDir, 0o700); err != nil {
+		return nil, fmt.Errorf("failed to create cache directory: %w", err)
+	}
+
+	fc := &FileCache[T]{
+		baseDir:       baseDir,
+		expiryMinutes: expiryMinutes,
+	}
+
+	// Clean up expired files synchronously
+	fc.cleanupExpiredFiles()
+
+	return fc, nil
+}
+
+// cleanupExpiredFiles removes expired cache files from disk.
+// This runs synchronously once when the cache is created.
+func (fc *FileCache[T]) cleanupExpiredFiles() {
+	now := time.Now()
+
+	_ = filepath.Walk(fc.baseDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil || info.IsDir() {
+			return nil
+		}
+
+		// Only process .json cache files
+		if filepath.Ext(info.Name()) != ".json" {
+			return nil
+		}
+
+		// Try to read the cache entry
+		data, err := os.ReadFile(path)
+		if err != nil {
+			return nil
+		}
+
+		var entry cacheEntry
+		if err := json.Unmarshal(data, &entry); err != nil {
+			// Delete corrupted files
+			_ = os.Remove(path)
```

Comment:
same as in other places, we should log the error, at least in debug, maybe warn.